### PR TITLE
Stricter date validation

### PR DESCRIPTION
**Pull request by seadragon2000341**
Current Date validation only checks if it is in the format YYYY-MM-DD

Added stricter date validation to make sure date is of the correct format and a valid date.

Also added more comprehensive testing for Date class

### CODE REVIEW

File: src/main/java/seedu/address/model/internship/Date.java [hunk 1]
```diff
@@ -32,7 +34,17 @@ public Date(String date) {
      * @returns true if a given string is a valid date string, else returns false.
      */
     public static boolean isValidDate(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        };
```
Comment (eugenetangkj) [inline src/main/java/seedu/address/model/internship/Date.java:39]: I think there is an additional semi-colon here. This causes a checkstyle violation:
image

As a tip, you could double press `ctrl` and run `gradlew checkstyleMain checkstyleTest` to ensure that there is no coding style violations.

**Review by eugenetangkj — COMMENTED** [review]
Very good effort. Simplistic way of testing for valid dates, and good test coverage. I tried it locally on my computer and it works. Just a minor fix before we can merge. 👍

File: src/main/java/seedu/address/model/internship/Date.java (hunk 1 above)
Comment (seadragon2000341) [reply src/main/java/seedu/address/model/internship/Date.java]: really helpful tip! thanks Eugene! Have done the changes

**Comment by eugenetangkj** [comment]
Looks good to merge!

**Pull request merged**